src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
```

Comment:
> Before this PR SettingsDAO has serialized data which where likely misinterpreted after an update and mixxx.cfg has human readable and editable data. I like the idea to treat the data as "sacred", but we don't do this with the header state.

Right -- header data is opaque since we can't really tell what Qt is storing in there (well, we could but we don't). So we rely on Qt to correctly save/restore it.  

Do you mean that the header data already contains sorting state so there's the possibility for this PR to store data conflicting with the header state? 

> What will be the rule of thump, which storage is used when?

I agree conceptually it's awkward to have two different places to store key/value configuration data. I think the rule of thumb is that the library settings table is for storing library-specific configuration data. It's nice that a user can copy their mixxxdb.sqlite to a different profile and still have a working library. So things like the current schema version absolutely have to go in SettingsDAO, but things like the preferred column sort order is less clear-cut. I'd be OK with the rule of thumb being "all library-specific preferences go in the library settings table".